### ISSUE TITLE
"afterNextPatch:true" is missing on "client.send()"

### ISSUE THREAD

**Issue by dengzhaofun** [colyseus/colyseus#304]
Does room.send also use the state change patch push strategy, because I expect the interface to modify data and send data in the same push?

**Comment by endel** [comment]
That's a good point, [`this.send()`](https://docs.colyseus.io/server/room/#send-client-message) does not currently have the `{afterNextPatch:true}` option that [`this.broadcast()`](https://docs.colyseus.io/server/room/#broadcast-message-options) has. 

Gonna add this option for `this.send()` soon!

**Comment by dengzhaofun** [comment]
hi @endel Do you mean that the synchronization of this.send and data state change may not be completed in the same patch, so whether the client receiving onMessage and onStateChange will result in data inconsistency?

**Comment by endel** [comment]
@dengzhaofun exactly, `this.send()` and `this.broadcast()` are sent immediatelly. Only `this.broadcast(data, { afterNextPatch: true })` will ensure the message will be sent after the next state patch

**Comment by dengzhaofun** [comment]
Ok, look forward to solving

**Comment by dengzhaofun** [comment]
Hi, @endel 
It is hoped that this issue can be resolved as soon as possible. Imagine a scenario where people in the room need to initiate a vote to end the game. Players vote to modify data on the state, and the final result of the vote is to notify the client through this.send. Then it is possible that this.send arrives at the client in advance, causing the game to end without seeing the voting status.

**Comment by endel** [comment]
Hi @dengzhaofun, supporting `afterNextPatch: true` on `send()` should be implemented soonish (next week probably)

You can handle this scenario without relying on sending messages though. When mutating the state, you can listen for a change on the particular variable you're interested on, and display it for all users.

**Example**

```typescript
/**
  * Server-side
  */
class VoteResults extends Schema {
  @type("string") result;
}

class GameState extends Schema {
  @type(VoteResults) voteResult = new VoteResults();
}
// ...
this.state.voteResult.result = "something";

/**
  * Client-side
  */
room.state.voteResult.listen("result", (value) => {
  console.log("Result is", value);
});
```

**Comment by endel** [comment]
Closing as this has been fixed on 0.14.18 🥳
